File: app/src/main/java/com/asfoundation/wallet/ui/iab/OnChainBuyView.java
package com.asfoundation.wallet.ui.iab;

import android.os.Bundle;
import io.reactivex.Observable;
import java.math.BigDecimal;
import java.util.List;

/**
 * Created by franciscocalado on 19/07/2018.
 */

public interface OnChainBuyView {

  Observable<Object> getOkErrorClick();

  void showLoading();

  void close(Bundle data);

  void finish(Bundle data);

  void showError();

  void showTransactionCompleted();

  void showWrongNetworkError();

  void showNoNetworkError();

  void showApproving();

  void showBuying();

  void showNonceError();

  void showNoTokenFundsError();

  void showNoEtherFundsError();

  void showNoFundsError();

  void showRaidenChannelValues(List<BigDecimal> values);

  long getAnimationDuration();
}

File: app/src/main/java/com/asfoundation/wallet/ui/iab/OnChainBuyView.kt
package com.asfoundation.wallet.ui.iab

import android.os.Bundle
import androidx.annotation.StringRes
import io.reactivex.Observable
import java.math.BigDecimal

interface OnChainBuyView {

  fun getOkErrorClick(): Observable<Any?>

  fun getSupportIconClick(): Observable<Any?>

  fun getSupportLogoClick(): Observable<Any?>

  fun close(data: Bundle?)

  fun finish(data: Bundle?, txId: String)

  fun showError(@StringRes message: Int? = null)

  fun showTransactionCompleted()

  fun showWrongNetworkError()

  fun showNoNetworkError()

  fun showApproving()

  fun showBuying()

  fun showNonceError()

  fun showNoTokenFundsError()

  fun showNoEtherFundsError()

  fun showNoFundsError()

  fun showForbiddenError()

  fun showRaidenChannelValues(values: List<BigDecimal>)

  fun getAnimationDuration(): Long

  fun lockRotation()

  fun showVerification()
}